Src/FluentAssertions/Xml/XDocumentAssertions.cs

```diff
@@ -171,14 +174,14 @@ public AndWhichConstraint<XDocumentAssertions, XElement> HaveRoot(XName expected
 
         XElement root = Subject.Root;
 
-        Execute.Assertion
+        assertionChain
             .ForCondition(root is not null && root.Name == expected)
             .BecauseOf(because, becauseArgs)
             .FailWith(
                 "Expected {context:subject} to have root element {0}{reason}, but found {1}.",
                 expected.ToString(), Subject);
 
-        return new AndWhichConstraint<XDocumentAssertions, XElement>(this, root);
+        return new AndWhich<XDocumentAssertions, XElement>(this, root, assertionChain, $"/{expected}");
```

Comment:
What about concatenating this with `.` like we access members in C#:

```csharp
var doc = ...
doc.Should().HaveRoot("member").Which...
```

which would then result in `Expected doc.member to have...`

Reply:
Because we're talking about XML paths, and they are usually displayed using `/` as a separator.

Reply:
Hmm.. I think we need something different than what nested `AssertionScope`s use when chaining their contexts together.

If I get an exception that says something like this: `Expected Foo/Bar/Baz/doc/member to have...` it wouldn't be entirely sure what maps to what. Do you see my concern?